mesonbuild/interpreter/type_checking.py

```diff
@@ -481,6 +481,11 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
     KwargInfo('verbose', bool, default=False, since='0.62.0'),
 ]
 
+_VS_MODULE_DEF_KW: KwargInfo[T.Union[str, File, CustomTarget, CustomTargetIndex]] = KwargInfo(
+    'vs_module_defs',
+    (str, File, CustomTarget, CustomTargetIndex, NoneType),
```

Comment:
Unfortunately, CustomTargetIndex does not work and never has worked. See #12053 and the linked issue.

Reply:
`CustomTargetIndex` has a `get_filename()` method though, and has since it was initially added, so I think it does actually work